### PR TITLE
Update matcher to better match layernorm fusions in other models

### PR DESCRIPTION
**Pull request by pfultz2**

### CODE REVIEW

**Comment by pfultz2** [comment]
This also needs unit test for the update to `find_inner_broadcast`.

**Comment by kahmed10** [comment]
PR #1491 modified the matcher to layernorm and broke MLPerf bert... This PR happens to fix the issue. Please prioritize for merging.

File: src/targets/gpu/include/migraphx/gpu/prefuse_ops.hpp [hunk 1]
```diff
@@ -30,14 +30,14 @@
...
-struct module;
+struct module_pass_manager;
 
 namespace gpu {
 
 struct prefuse_ops
 {
     std::string name() const { return "gpu::prefuse_ops"; }
-    void apply(module& m) const;
+    void apply(module_pass_manager& m) const;
```
Comment (kahmed10) [inline src/targets/gpu/include/migraphx/gpu/prefuse_ops.hpp:40]:
```suggestion
    void apply(module_pass_manager& mpm) const;
```

**Review by kahmed10 — APPROVED** [review]

**Comment by pfultz2** [comment]
This is still missing unit tests for `find_inner_broadcast` change, I am not sure I will get that done this weekend.

**Comment by pfultz2** [comment]
I added a test.

**Review by TedThemistokleous — APPROVED** [review]

**Pull request merged**